todocore/src/main/java/com/knighttodo/todocore/validation/validator/DateValidator.java

```diff
@@ -0,0 +1,40 @@
+package com.knighttodo.todocore.validation.validator;
+
+import com.knighttodo.todocore.validation.annotation.ValidDate;
+import lombok.extern.slf4j.Slf4j;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+@Slf4j
+public class DateValidator implements ConstraintValidator<ValidDate, String> {
+
+    private Pattern pattern;
+
+    @Override
+    public void initialize(ValidDate validDate) {
+        try {
+            pattern = Pattern.compile(validDate.regexp());
+        } catch (PatternSyntaxException ex) {
+            throw new IllegalArgumentException("Given regex is invalid", ex);
```

Comment:
Are you validating regex here or a date format?

Reply:
I am validating regex, if regex doesn't compile I throw exception

Reply:
I'm not sure that IllegalArgumentException is suited here then. I think that it should be a specific exception for it. So it's a check for the correct regexp pattern so it won't be broken in the runtime. And we can use different patterns afterward but why not create a new annotation for a different pattern next time? And are we going to need it in the future at all? It looks like overengineering to me, though it's an interesting thing, though you should understand that you shouldn't do it, you need minimum needed implementation, and everything else is additional code to test, maintain, discuss, and read.

Reply:
ok